upgrade/upgrade-scripts/upgrade

```diff
@@ -395,16 +397,21 @@ function finalize() {
 		;;
 	esac
 
-	ROOTFS_DATASET=$(get_mounted_rootfs_container_dataset)
-	[[ -n "$ROOTFS_DATASET" ]] ||
-		die "unable to determine mounted rootfs container dataset"
-
-	SNAPSHOT_NAME=$(get_dataset_rollback_snapshot_name "$ROOTFS_DATASET")
-	[[ -n "$SNAPSHOT_NAME" ]] ||
-		die "unable to determine rollback snapshot name"
-
-	zfs destroy -r "$ROOTFS_DATASET@$SNAPSHOT_NAME" ||
-		die "failed to destroy rollback snapshots"
+	SNAPSHOT_LIST=$(zfs list -t snapshot -r rpool/ROOT -d 2 -Ho name) ||
+		die "unable to determine available rollback snapshots"
+
+	for snapshot in $SNAPSHOT_LIST; do
+		local SNAPSHOT_VERSION
+		SNAPSHOT_VERSION=$(zfs get -Hpo value "$PROP_CURRENT_VERSION" "$snapshot") ||
+			die "failed to get snapshot version"
+		compare_versions \
+			"$SNAPSHOT_VERSION" "lt" "$UPGRADE_BASE_VERSION"
+		local rc="$?"
+		if [[ "$rc" == 0 ]]; then
```

Comment:
Could we remove the `rc` variable here, and more simply do this:
```
if [[ "$?" == 0 ]]; then
```
it's not a big deal, but I don't see where `rc` is used elsewhere, so it might be reasonable to remove it..

Along the same lines, we could probably do this too:
```
if compare_versions \
    "$SNAPSHOT_VERSION" "lt" "$UPGRADE_BASE_VERSION"; then
```
IMO, this would be best, since it would remove our use of `$?` as well.. the use of `$?` can be problematic if code is mistakenly added (later) in between the call to `compare_versions` and the use of `$?`.

Reply:
Will do, that's a good point